dags/tests/test_property_definitions.py

```diff
@@ -0,0 +1,236 @@
+    cases = {
+        UUID(int=i): case
+        for i, case in enumerate(
+            [
+                PropertyTypeTestData("{}", []),
+                # special cases: key patterns
+                PropertyTypeTestData('{"utm_source": 123}', [("utm_source", "String")]),
+                PropertyTypeTestData('{"$feature/a": false}', [("$feature/a", "String")]),
+                PropertyTypeTestData(
+                    '{"$survey_response": 1, "$survey_response_2": 2}',
+                    [("$survey_response", "String"), ("$survey_response_2", "String")],
+                ),
+                # special cases: timestamp detection
+                PropertyTypeTestData(
+                    json.dumps({"timestamp": time.time()}),
+                    [("timestamp", "DateTime")],  # XXX: can't be parsed back out correctly!
+                ),
+                PropertyTypeTestData(json.dumps({"timestamp": int(time.time())}), [("timestamp", "DateTime")]),
+                PropertyTypeTestData(
+                    json.dumps({"timestamp": time.time() - timedelta(days=365).total_seconds()}),
+                    [("timestamp", "Numeric")],  # "too old"
+                ),
+                # special cases: string values
+                PropertyTypeTestData('{"a": "true"}', [("a", "Boolean")]),
+                PropertyTypeTestData('{"a": "TRUE"}', [("a", "Boolean")]),
+                PropertyTypeTestData('{"a": "false"}', [("a", "Boolean")]),
+                PropertyTypeTestData('{"a": "FALSE"}', [("a", "Boolean")]),
+                PropertyTypeTestData('{"a": " true "}', [("a", "Boolean")]),
+                PropertyTypeTestData(
+                    '{"a": "true\\n"}',
+                    [("a", "String")],  # XXX: inconsistent with existing implementation
+                ),
+                PropertyTypeTestData(  # weird formats clickhouse allows
+                    '{"a": "23:21:04", "b": "September", "c": "2024", "d": "1"}',
+                    [("a", "String"), ("b", "String"), ("c", "String"), ("d", "String")],
+                ),
+                PropertyTypeTestData('{"a": "2025-04-24"}', [("a", "DateTime")]),
+                PropertyTypeTestData('{"a": "04/24/2025"}', [("a", "DateTime")]),
+                PropertyTypeTestData(
+                    '{"a": "2025-04-24 23:21:04"}',  # ISO-8601, space separator, no tz
+                    [("a", "DateTime")],
+                ),
+                PropertyTypeTestData(
+                    '{"a": "2025-04-24T23:21:04+0000"}',  # ISO-8601, t separator, with tz
+                    [("a", "DateTime")],
+                ),
+                PropertyTypeTestData(
+                    '{"a": "2025-04-24T23:21:04+00:00"}',  # RFC 3339
+                    [("a", "DateTime")],
+                ),
+                PropertyTypeTestData(
+                    '{"a": "Thu, 24 Apr 2025 23:21:04 +0000"}',  # RFC 2822
+                    [("a", "DateTime")],
+                ),
```

Comment:
This is generally consistent with https://github.com/PostHog/posthog/blob/052f4ea40c5043909115f835f09445e18dd9727c/rust/property-defs-rs/src/types.rs#L47-L51 but in some cases a little stricter, in others a little looser

But whatever is parsed into a `DateTime` here can actually be read _as_ a `DateTime` by ClickHouse which wasn't necessarily true previously